src/GraphQL.SystemTextJson/GraphQLSerializer.cs

```diff
@@ -134,6 +136,26 @@ private void ConfigureOptions(IErrorInfoProvider errorInfoProvider)
             {
                 _options.Converters.Add(new JsonConverterBigInteger());
             }
+
+            if (!_options.Converters.Any(c => c.CanConvert(typeof(Inputs))))
+            {
+                _options.Converters.Add(new InputsJsonConverter());
+            }
+
+            if (!_options.Converters.Any(c => c.CanConvert(typeof(GraphQLRequest))))
+            {
+                _options.Converters.Add(new GraphQLRequestJsonConverter());
+            }
+
+            if (!_options.Converters.Any(c => c.CanConvert(typeof(List<GraphQLRequest>))))
```

Comment:
🤔 Do not the lists are detected by serializer internals automatically so serializer can use registered converter for list element?

Reply:
Yes.  But then it will not decode `{"query":"test"}` into a `GraphQLRequest[]`.  Rather it will say that an array was expected but an object was provided.  So it was necessary for custom list converters for that type of decoding.

As one of the goals here is to support typical HTTP/JSON transports in use, and as one of the needs is to decode a single OR multiple requests, I added custom converters for `GraphQLRequest` list types.

Reply:
>  But then it will not decode {"query":"test"} into a GraphQLRequest[]

Where is it necessary?

Reply:
See:

https://github.com/graphql-dotnet/server/blob/master/src/Transports.AspNetCore.SystemTextJson/GraphQLRequestDeserializer.cs#L50..L68

As you can see it can decode a single or batch request.  My design is such that attempting to deserialize a `GraphQLRequest[]` (or other enumerable type) will successfully parse a single or batch request.

Or deserialize a `GraphQLRequest` for single-request-only support.

Reply:
To clarify your intentions in server repo. Do you want to always deserialize into `GraphQLRequest[]` type and then check length?

Reply:
Yes